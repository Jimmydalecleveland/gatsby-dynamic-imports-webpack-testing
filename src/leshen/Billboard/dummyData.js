import lifestyle from '../../images/split-content-lifestyle.jpg'
import lifestyleFull from '../../images/billboard-lifestyle-full.jpg'

export const FLUID_IMAGE_DATA = {
  context: {
    custom: {
      alt: 'Alt Text',
    },
  },
  fluid: {
    base64:
      'data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAPABQDASIAAhEBAxEB/8QAGAAAAwEBAAAAAAAAAAAAAAAAAAQFAQL/xAAVAQEBAAAAAAAAAAAAAAAAAAABAv/aAAwDAQACEAMQAAABY4VxKBIGf//EABoQAAIDAQEAAAAAAAAAAAAAAAABAgMREjL/2gAIAQEAAQUCrjpP1qE3lksOz//EABYRAQEBAAAAAAAAAAAAAAAAAAAREv/aAAgBAwEBPwHSv//EABURAQEAAAAAAAAAAAAAAAAAAAAR/9oACAECAQE/AYj/xAAcEAACAQUBAAAAAAAAAAAAAAAAATECEBJBUWH/2gAIAQEABj8C6JUySZ6F7b//xAAcEAACAgIDAAAAAAAAAAAAAAABEQAxIVEQQZH/2gAIAQEAAT8hIHQaigsQF4gXT2NWQVuEdhW3x//aAAwDAQACAAMAAAAQ4+//xAAXEQADAQAAAAAAAAAAAAAAAAAAAREh/9oACAEDAQE/EMVUo//EABYRAQEBAAAAAAAAAAAAAAAAAAARUf/aAAgBAgEBPxDaH//EAB4QAQACAgEFAAAAAAAAAAAAAAERIQAxcUFRYZGx/9oACAEBAAE/EEKI60I86xyRZUEjqPvNJHwy9miG2LXgZRab+9sUtAHGf//Z',
    src: lifestyle,
    srcSet: lifestyle,
    aspectRatio: 1.3524904214559388,
    sizes: '(max-width: 706px) 100vw, 706px',
    presentationWidth: 2560,
  },
}

export const FLUID_IMAGE_DATA_FULL = {
  context: {
    custom: {
      alt: 'Alt Text',
    },
  },
  fluid: {
    base64:
      'data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAGABQDASIAAhEBAxEB/8QAFgABAQEAAAAAAAAAAAAAAAAAAAMC/8QAFQEBAQAAAAAAAAAAAAAAAAAAAAH/2gAMAwEAAhADEAAAAdULJA//xAAZEAACAwEAAAAAAAAAAAAAAAABAgADBBL/2gAIAQEAAQUCTYTBa1kfXZ3/AP/EABQRAQAAAAAAAAAAAAAAAAAAABD/2gAIAQMBAT8BP//EABURAQEAAAAAAAAAAAAAAAAAABAR/9oACAECAQE/AYf/xAAcEAACAQUBAAAAAAAAAAAAAAAAEQECEiExQXH/2gAIAQEABj8C0i6ZxTwlJeH/xAAZEAADAQEBAAAAAAAAAAAAAAABESEAQVH/2gAIAQEAAT8hN23nNcLWQCawQJjHn//aAAwDAQACAAMAAAAQBB//xAAWEQEBAQAAAAAAAAAAAAAAAAABEBH/2gAIAQMBAT8QXJ//xAAWEQEBAQAAAAAAAAAAAAAAAAABABH/2gAIAQIBAT8QA2X/xAAaEAEAAgMBAAAAAAAAAAAAAAABACERMWGR/9oACAEBAAE/EGQY0bPrcwQjrko72algtuO3P//Z',
    src: lifestyleFull,
    srcSet: lifestyleFull,
    aspectRatio: 3.116883116883117,
    sizes: '(max-width: 800px) 100vw, 800px',
    presentationWidth: 2560,
  },
}
