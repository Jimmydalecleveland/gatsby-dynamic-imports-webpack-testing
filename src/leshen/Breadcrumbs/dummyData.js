export const CRUMB_DATA = [
  {
    shortName: 'Home',
    path: '/',
  },
  {
    shortName: 'Internet',
    path: '/internet',
  },
  {
    shortName: 'Gigabit',
    path: '/gigabit',
  },
  {
    shortName: 'Gigabit Plans',
    path: '/gigabit-plans',
  },
]
