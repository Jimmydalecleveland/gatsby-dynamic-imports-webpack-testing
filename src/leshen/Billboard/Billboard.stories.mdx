import {
  Story,
  Meta,
  Preview,
  Props,
  Anchor,
} from "@storybook/addon-docs/blocks"

import Billboard, { ticketID } from "."
import Button from "../Button"
import Typography from "../Typography"
import Link from "../Link"
import ComponentProgress from "../ComponentProgress"
import BillboardImage from "../../images/contentful-stories/billboard.png"
import { BUTTON } from "../utils/dataTrackingValues"

import { FLUID_IMAGE_DATA, FLUID_IMAGE_DATA_FULL } from "./dummyData"

<Meta
  title="Components|Billboard"
  component={Billboard}
  decorators={[
    story => (
      <>
        <ComponentProgress ticketID={ticketID} />
        {story()}
      </>
    ),
  ]}
/>

# Billboard

<Preview>
  <Story name="basic">
    <Billboard
      heading="Price for Life"
      mainContent={
        <>
          <Typography>
            CenturyLink Price for Life Internet Plans ensure your monthly
            internet rate stays the same when you remain on the same internet
            plan at the same address. No contract, no rate hikes.
          </Typography>
          <Typography>
            See what CenturyLink Services are in your area:{" "}
            <Link to="/section-link"> Call Now</Link>
          </Typography>
          <Button primary data-tracking={BUTTON}>
            Submit
          </Button>
        </>
      }
      image={FLUID_IMAGE_DATA}
    />
  </Story>
</Preview>

## Props

<Props of={Billboard} />

## Dark Content Container

<Preview>
  <Story name="dark">
    <Billboard
      backgroundColor="Dark"
      heading="Price for Life"
      mainContent={
        <>
          <Typography>
            CenturyLink Price for Life Internet Plans ensure your monthly
            internet rate stays the same when you remain on the same internet
            plan at the same address. No contract, no rate hikes.
          </Typography>
          <Typography>
            See what CenturyLink Services are in your area:{" "}
            <Link to="/section-link"> Call Now</Link>
          </Typography>
          <Button>Submit</Button>
        </>
      }
      image={FLUID_IMAGE_DATA}
    />
  </Story>
</Preview>

## Prominent Content Container

<Preview>
  <Story name="prominent">
    <Billboard
      backgroundColor="Prominent"
      heading="Price for Life"
      mainContent={
        <>
          <Typography>
            CenturyLink Price for Life Internet Plans ensure your monthly
            internet rate stays the same when you remain on the same internet
            plan at the same address. No contract, no rate hikes.
          </Typography>
          <Typography>
            See what CenturyLink Services are in your area:{" "}
            <Link to="/section-link"> Call Now</Link>
          </Typography>
          <Button primary data-tracking={BUTTON}>
            Submit
          </Button>
        </>
      }
      image={FLUID_IMAGE_DATA}
    />
  </Story>
</Preview>

## Full Width Background

When the `variant` prop is passed `'full'`, the background image will take up the full width of it's container.

<Preview>
  <Story name="full width background">
    <Billboard
      variant="full"
      heading={<Typography variant="h2">Price for Life</Typography>}
      mainContent={
        <>
          <Typography>
            CenturyLink Price for Life Internet Plans ensure your monthly
            internet rate stays the same when you remain on the same internet
            plan at the same address. No contract, no rate hikes.
          </Typography>
          <Typography>
            See what CenturyLink Services are in your area:{" "}
            <Link to="/section-link"> Call Now</Link>
          </Typography>
          <Button primary data-tracking={BUTTON}>
            Submit
          </Button>
        </>
      }
      image={FLUID_IMAGE_DATA_FULL}
    />
    <Billboard
      variant="full"
      prominent
      heading={<Typography variant="h2">Price for Life</Typography>}
      mainContent={
        <>
          <Typography>
            CenturyLink Price for Life Internet Plans ensure your monthly
            internet rate stays the same when you remain on the same internet
            plan at the same address. No contract, no rate hikes.
          </Typography>
          <Typography>
            See what CenturyLink Services are in your area:
          </Typography>
          <Button primary data-tracking={BUTTON}>
            Submit
          </Button>
        </>
      }
      image={FLUID_IMAGE_DATA_FULL}
    />
  </Story>
</Preview>

## Contentful

<Story name="contentful">
  <div>
    <p>
      In a "Section" entry, select "Billboard" from the "type" selector. The
      entire section will become a Billboard component.
      <strong>
        The "Identifier" and "Featured Image" fields are required for this
        entry.
      </strong>
    </p>
    <img src={BillboardImage} alt="billboard type selector in contentful" />
  </div>
</Story>
