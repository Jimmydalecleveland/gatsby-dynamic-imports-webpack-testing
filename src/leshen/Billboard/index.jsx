import React from 'react'
import PropTypes from 'prop-types'

import Typography from '../Typography'
import * as Styled from './Billoard.styles'
import Placeholder from '../Placeholder'
import Breadcrumbs from '../Breadcrumbs'

const Billboard = ({
  image,
  heading,
  subheading,
  mainContent,
  backgroundColor,
  breadcrumbProps,
  variant,
  children,
  headingLevel,
  ...props
}) => (
  <Styled.Section
    data-testid="Billboard-Section"
    backgroundColor={backgroundColor}
    {...props}
  >
    {breadcrumbProps && (
      <Breadcrumbs {...breadcrumbProps} backgroundColor={backgroundColor} />
    )}
    <Styled.Content
      data-testid="Billboard-Content"
      backgroundColor={backgroundColor}
      variant={variant}
    >
      <Typography variant={`h${headingLevel}`}>{heading}</Typography>
      {subheading && (
        <Typography
          variant={`h${headingLevel + 3}`}
          rendersAs={`h${headingLevel + 1}`}
        >
          {subheading}
        </Typography>
      )}
      {mainContent}
      {children}
    </Styled.Content>
    {image ? (
      <Styled.Image data={image} variant={variant} maxWidth="none" />
    ) : (
      <Placeholder type="error">
        <Typography variant="small">Notice</Typography>
        <Typography>
          The content entry of type &quot;Billboard&quot; is missing the Image
          media asset.
        </Typography>
      </Placeholder>
    )}
  </Styled.Section>
)

Billboard.defaultProps = {
  variant: 'split',
  backgroundColor: null,
  subheading: null,
  mainContent: null,
  children: null,
  breadcrumbProps: null,
  headingLevel: 2,
}

Billboard.propTypes = {
  /** Expects the data from ImageSharp for an image */
  image: PropTypes.shape({
    fluid: PropTypes.object,
    fixed: PropTypes.object,
    svg: PropTypes.object,
    context: PropTypes.object,
  }).isRequired,
  variant: PropTypes.oneOf(['split', 'full']),
  /** Sets the content background to the themes "primary" color. */
  backgroundColor: PropTypes.string,
  heading: PropTypes.string.isRequired,
  subheading: PropTypes.string,
  mainContent: PropTypes.node,
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.objectOf(PropTypes.node),
    PropTypes.arrayOf(PropTypes.node),
  ]),
  breadcrumbProps: PropTypes.shape({
    crumbs: PropTypes.string,
    path: PropTypes.string,
    siteURL: PropTypes.string,
  }),
  headingLevel: PropTypes.oneOf([1, 2]),
}

export default Billboard

export const ticketID = 'CMS-156'
