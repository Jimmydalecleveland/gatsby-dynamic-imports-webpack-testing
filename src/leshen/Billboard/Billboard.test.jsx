import React from "react"
import { render } from "setup-tests"

import Billboard from "."
import { FLUID_IMAGE_DATA } from "./dummyData"
import theme from "../utils/theme"

describe("Billboard", () => {
  test("should render and have content", () => {
    const { container, getByTestId } = render(
      <Billboard
        image={FLUID_IMAGE_DATA}
        heading="Heading"
        subheading="Subheading"
        mainContent={
          <>
            <p>Main Content</p>
          </>
        }
      />
    )
    expect(container).toMatchSnapshot()
    const contentContainer = getByTestId("Billboard-Content")
    expect(contentContainer).toHaveTextContent("Heading")
    expect(contentContainer).toHaveTextContent("Subheading")
    expect(contentContainer).toHaveTextContent("Main Content")
  })

  test("should render full variant", () => {
    const { container } = render(
      <Billboard
        image={FLUID_IMAGE_DATA}
        variant="full"
        heading="Heading"
        subheading="Subheading"
        mainContent={
          <>
            <p>Main Content</p>
          </>
        }
      />
    )
    expect(container).toMatchSnapshot()
  })

  test("should render prominent, full variant", () => {
    // rerender doesn't work for styles as expected. Otherwise this
    // should probably be in the previous test as a rerender of props.
    const { getByTestId } = render(
      <Billboard
        image={FLUID_IMAGE_DATA}
        backgroundColor="Prominent"
        variant="full"
        heading="Heading"
        subheading="Subheading"
        mainContent={
          <>
            <p>Main Content</p>
          </>
        }
      />
    )

    const sectionContainer = getByTestId("Billboard-Section")
    const contentContainer = getByTestId("Billboard-Content")
    expect(contentContainer).toMatchSnapshot()
    expect(sectionContainer).toHaveStyle(`
        color: #fff;
        background-color: ${theme.colors[theme.prominent].base.hex};
    `)
    expect(contentContainer).toHaveStyle(`
        color: #fff;
    `)
  })
})
