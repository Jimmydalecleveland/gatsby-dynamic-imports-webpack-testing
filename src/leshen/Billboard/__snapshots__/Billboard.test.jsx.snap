// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Billboard should render and have content 1`] = `
.emotion-10 {
  padding-left: calc(2.2222222222222223vw + 21.333333333333332px);
  padding-right: calc(2.2222222222222223vw + 21.333333333333332px);
  padding-top: calc(7.777777777777778vw + -5.333333333333333px);
  padding-bottom: calc(7.777777777777778vw + -5.333333333333333px);
  position: relative;
  position: relative;
}

@media (max-width:480px) {
  .emotion-10 {
    padding-left: 32px;
  }
}

@media (min-width:1200px) {
  .emotion-10 {
    padding-left: 48px;
  }
}

@media (max-width:480px) {
  .emotion-10 {
    padding-right: 32px;
  }
}

@media (min-width:1200px) {
  .emotion-10 {
    padding-right: 48px;
  }
}

@media (min-width:1288px) {
  .emotion-10 {
    padding-right: calc((100% - 1200px) / 2);
    padding-left: calc((100% - 1200px) / 2);
  }
}

@media (max-width:480px) {
  .emotion-10 {
    padding-top: 32px;
  }
}

@media (min-width:1200px) {
  .emotion-10 {
    padding-top: 88px;
  }
}

@media (max-width:480px) {
  .emotion-10 {
    padding-bottom: 32px;
  }
}

@media (min-width:1200px) {
  .emotion-10 {
    padding-bottom: 88px;
  }
}

.emotion-4 {
  position: relative;
  z-index: 1;
}

@media (min-width:768px) {
  .emotion-4 {
    width: 50%;
    box-shadow: none;
    padding-right: calc(4.444444444444445vw + 10.666666666666666px);
  }

@media (max-width:480px) {
    .emotion-4 {
      padding-right: 32px;
    }
}

@media (min-width:1200px) {
    .emotion-4 {
      padding-right: 64px;
    }
}
}

@media (min-width:768px) {
  section:nth-of-type(even) .emotion-4 {
    left: 50%;
    padding-left: calc(4.444444444444445vw + 10.666666666666666px);
    padding-right: 0;
  }

@media (max-width:480px) {
    section:nth-of-type(even) .emotion-4 {
      padding-left: 32px;
    }
}

@media (min-width:1200px) {
    section:nth-of-type(even) .emotion-4 {
      padding-left: 64px;
    }
}
}

.emotion-0 {
  font-weight: 800;
  font-size: calc(2.7777777777777777vw + 14.666666666666666px);
  line-height: calc(3.3333333333333335vw + 16px);
}

@media (max-width:480px) {
  .emotion-0 {
    font-size: 28px;
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    font-size: 48px;
  }
}

@media (max-width:480px) {
  .emotion-0 {
    line-height: 32px;
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    line-height: 56px;
  }
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-0 {
    width: 100%;
  }
}

.emotion-2 {
  font-weight: 500;
  font-size: calc(0.5555555555555556vw + 17.333333333333332px);
  line-height: calc(1.1111111111111112vw + 18.666666666666668px);
}

@media (max-width:480px) {
  .emotion-2 {
    font-size: 20px;
  }
}

@media (min-width:1200px) {
  .emotion-2 {
    font-size: 24px;
  }
}

@media (max-width:480px) {
  .emotion-2 {
    line-height: 24px;
  }
}

@media (min-width:1200px) {
  .emotion-2 {
    line-height: 32px;
  }
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-2 {
    width: 100%;
  }
}

.emotion-7 {
  max-width: none;
  width: 100%;
  margin: 0 auto;
  display: none;
  width: 50%;
}

@media (min-width:768px) {
  .emotion-7 {
    display: block;
    position: absolute !important;
    top: 0;
    right: 0;
    bottom: 0;
    left: 50%;
  }
}

@media (min-width:768px) {
  section:nth-of-type(even) .emotion-7 {
    -webkit-transform: scale(-1,1);
    -ms-transform: scale(-1,1);
    transform: scale(-1,1);
    right: 50%;
    left: 0;
  }
}

<div>
  <section
    class="emotion-9 emotion-10 emotion-11"
    data-testid="Billboard-Section"
  >
    <div
      class="emotion-4 emotion-5"
      data-testid="Billboard-Content"
    >
      <h2
        class="emotion-0 emotion-1"
      >
        Heading
      </h2>
      <h3
        class="emotion-2 emotion-3"
      >
        Subheading
      </h3>
      <p>
        Main Content
      </p>
    </div>
    <div
      class="emotion-6 emotion-7 emotion-8 gatsby-image-wrapper"
      style="position: relative; overflow: hidden;"
    >
      <div
        aria-hidden="true"
        style="width: 100%; padding-bottom: 73.93767705382436%;"
      />
      <img
        alt=""
        aria-hidden="true"
        src="data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAPABQDASIAAhEBAxEB/8QAGAAAAwEBAAAAAAAAAAAAAAAAAAQFAQL/xAAVAQEBAAAAAAAAAAAAAAAAAAABAv/aAAwDAQACEAMQAAABY4VxKBIGf//EABoQAAIDAQEAAAAAAAAAAAAAAAABAgMREjL/2gAIAQEAAQUCrjpP1qE3lksOz//EABYRAQEBAAAAAAAAAAAAAAAAAAAREv/aAAgBAwEBPwHSv//EABURAQEAAAAAAAAAAAAAAAAAAAAR/9oACAECAQE/AYj/xAAcEAACAQUBAAAAAAAAAAAAAAAAATECEBJBUWH/2gAIAQEABj8C6JUySZ6F7b//xAAcEAACAgIDAAAAAAAAAAAAAAABEQAxIVEQQZH/2gAIAQEAAT8hIHQaigsQF4gXT2NWQVuEdhW3x//aAAwDAQACAAMAAAAQ4+//xAAXEQADAQAAAAAAAAAAAAAAAAAAAREh/9oACAEDAQE/EMVUo//EABYRAQEBAAAAAAAAAAAAAAAAAAARUf/aAAgBAgEBPxDaH//EAB4QAQACAgEFAAAAAAAAAAAAAAERIQAxcUFRYZGx/9oACAEBAAE/EEKI60I86xyRZUEjqPvNJHwy9miG2LXgZRab+9sUtAHGf//Z"
        style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 1; transition-delay: 500ms;"
      />
      <picture>
        <source
          sizes="(max-width: 706px) 100vw, 706px"
          srcset="test-file-stub"
        />
        <img
          alt="Alt Text"
          loading="lazy"
          sizes="(max-width: 706px) 100vw, 706px"
          src="test-file-stub"
          srcset="test-file-stub"
          style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
        />
      </picture>
      <noscript>
        &lt;picture&gt;&lt;source srcset="test-file-stub" sizes="(max-width: 706px) 100vw, 706px" /&gt;&lt;img loading="lazy" sizes="(max-width: 706px) 100vw, 706px" srcset="test-file-stub" src="test-file-stub" alt="Alt Text" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
      </noscript>
    </div>
  </section>
</div>
`;

exports[`Billboard should render full variant 1`] = `
.emotion-10 {
  padding-left: calc(2.2222222222222223vw + 21.333333333333332px);
  padding-right: calc(2.2222222222222223vw + 21.333333333333332px);
  padding-top: calc(7.777777777777778vw + -5.333333333333333px);
  padding-bottom: calc(7.777777777777778vw + -5.333333333333333px);
  position: relative;
  position: relative;
}

@media (max-width:480px) {
  .emotion-10 {
    padding-left: 32px;
  }
}

@media (min-width:1200px) {
  .emotion-10 {
    padding-left: 48px;
  }
}

@media (max-width:480px) {
  .emotion-10 {
    padding-right: 32px;
  }
}

@media (min-width:1200px) {
  .emotion-10 {
    padding-right: 48px;
  }
}

@media (min-width:1288px) {
  .emotion-10 {
    padding-right: calc((100% - 1200px) / 2);
    padding-left: calc((100% - 1200px) / 2);
  }
}

@media (max-width:480px) {
  .emotion-10 {
    padding-top: 32px;
  }
}

@media (min-width:1200px) {
  .emotion-10 {
    padding-top: 88px;
  }
}

@media (max-width:480px) {
  .emotion-10 {
    padding-bottom: 32px;
  }
}

@media (min-width:1200px) {
  .emotion-10 {
    padding-bottom: 88px;
  }
}

.emotion-0 {
  font-weight: 800;
  font-size: calc(2.7777777777777777vw + 14.666666666666666px);
  line-height: calc(3.3333333333333335vw + 16px);
}

@media (max-width:480px) {
  .emotion-0 {
    font-size: 28px;
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    font-size: 48px;
  }
}

@media (max-width:480px) {
  .emotion-0 {
    line-height: 32px;
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    line-height: 56px;
  }
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-0 {
    width: 100%;
  }
}

.emotion-2 {
  font-weight: 500;
  font-size: calc(0.5555555555555556vw + 17.333333333333332px);
  line-height: calc(1.1111111111111112vw + 18.666666666666668px);
}

@media (max-width:480px) {
  .emotion-2 {
    font-size: 20px;
  }
}

@media (min-width:1200px) {
  .emotion-2 {
    font-size: 24px;
  }
}

@media (max-width:480px) {
  .emotion-2 {
    line-height: 24px;
  }
}

@media (min-width:1200px) {
  .emotion-2 {
    line-height: 32px;
  }
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-2 {
    width: 100%;
  }
}

.emotion-4 {
  position: relative;
  z-index: 1;
  background-color: inherit;
}

@media (min-width:768px) {
  .emotion-4 {
    width: 50%;
    box-shadow: 0 0.3rem 1rem 0 rgba(0,0,0,0.1);
    padding: calc(2.2222222222222223vw + 5.333333333333333px);
    background-color: #fff;
  }

@media (max-width:480px) {
    .emotion-4 {
      padding: 16px;
    }
}

@media (min-width:1200px) {
    .emotion-4 {
      padding: 32px;
    }
}
}

@media (min-width:768px) {
  section:nth-of-type(even) .emotion-4 {
    left: 50%;
  }
}

.emotion-7 {
  max-width: none;
  width: 100%;
  margin: 0 auto;
  display: none;
}

@media (min-width:768px) {
  .emotion-7 {
    display: block;
    position: absolute !important;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
  }
}

@media (min-width:768px) {
  section:nth-of-type(even) .emotion-7 {
    -webkit-transform: scale(-1,1);
    -ms-transform: scale(-1,1);
    transform: scale(-1,1);
    right: 0;
    left: 0;
  }
}

<div>
  <section
    class="emotion-9 emotion-10 emotion-11"
    data-testid="Billboard-Section"
  >
    <div
      class="emotion-4 emotion-5"
      data-testid="Billboard-Content"
    >
      <h2
        class="emotion-0 emotion-1"
      >
        Heading
      </h2>
      <h3
        class="emotion-2 emotion-3"
      >
        Subheading
      </h3>
      <p>
        Main Content
      </p>
    </div>
    <div
      class="emotion-6 emotion-7 emotion-8 gatsby-image-wrapper"
      style="position: relative; overflow: hidden;"
    >
      <div
        aria-hidden="true"
        style="width: 100%; padding-bottom: 73.93767705382436%;"
      />
      <img
        alt=""
        aria-hidden="true"
        src="data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAPABQDASIAAhEBAxEB/8QAGAAAAwEBAAAAAAAAAAAAAAAAAAQFAQL/xAAVAQEBAAAAAAAAAAAAAAAAAAABAv/aAAwDAQACEAMQAAABY4VxKBIGf//EABoQAAIDAQEAAAAAAAAAAAAAAAABAgMREjL/2gAIAQEAAQUCrjpP1qE3lksOz//EABYRAQEBAAAAAAAAAAAAAAAAAAAREv/aAAgBAwEBPwHSv//EABURAQEAAAAAAAAAAAAAAAAAAAAR/9oACAECAQE/AYj/xAAcEAACAQUBAAAAAAAAAAAAAAAAATECEBJBUWH/2gAIAQEABj8C6JUySZ6F7b//xAAcEAACAgIDAAAAAAAAAAAAAAABEQAxIVEQQZH/2gAIAQEAAT8hIHQaigsQF4gXT2NWQVuEdhW3x//aAAwDAQACAAMAAAAQ4+//xAAXEQADAQAAAAAAAAAAAAAAAAAAAREh/9oACAEDAQE/EMVUo//EABYRAQEBAAAAAAAAAAAAAAAAAAARUf/aAAgBAgEBPxDaH//EAB4QAQACAgEFAAAAAAAAAAAAAAERIQAxcUFRYZGx/9oACAEBAAE/EEKI60I86xyRZUEjqPvNJHwy9miG2LXgZRab+9sUtAHGf//Z"
        style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 1; transition-delay: 500ms;"
      />
      <picture>
        <source
          sizes="(max-width: 706px) 100vw, 706px"
          srcset="test-file-stub"
        />
        <img
          alt="Alt Text"
          loading="lazy"
          sizes="(max-width: 706px) 100vw, 706px"
          src="test-file-stub"
          srcset="test-file-stub"
          style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
        />
      </picture>
      <noscript>
        &lt;picture&gt;&lt;source srcset="test-file-stub" sizes="(max-width: 706px) 100vw, 706px" /&gt;&lt;img loading="lazy" sizes="(max-width: 706px) 100vw, 706px" srcset="test-file-stub" src="test-file-stub" alt="Alt Text" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
      </noscript>
    </div>
  </section>
</div>
`;

exports[`Billboard should render prominent, full variant 1`] = `
.emotion-0 {
  font-weight: 800;
  font-size: calc(2.7777777777777777vw + 14.666666666666666px);
  line-height: calc(3.3333333333333335vw + 16px);
}

@media (max-width:480px) {
  .emotion-0 {
    font-size: 28px;
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    font-size: 48px;
  }
}

@media (max-width:480px) {
  .emotion-0 {
    line-height: 32px;
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    line-height: 56px;
  }
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-0 {
    width: 100%;
  }
}

.emotion-2 {
  font-weight: 500;
  font-size: calc(0.5555555555555556vw + 17.333333333333332px);
  line-height: calc(1.1111111111111112vw + 18.666666666666668px);
}

@media (max-width:480px) {
  .emotion-2 {
    font-size: 20px;
  }
}

@media (min-width:1200px) {
  .emotion-2 {
    font-size: 24px;
  }
}

@media (max-width:480px) {
  .emotion-2 {
    line-height: 24px;
  }
}

@media (min-width:1200px) {
  .emotion-2 {
    line-height: 32px;
  }
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-2 {
    width: 100%;
  }
}

.emotion-4 {
  color: #fff;
  background-color: #383d43 !important;
  position: relative;
  z-index: 1;
}

.emotion-4 * {
  color: #fff;
}

.emotion-4 a:link,
.emotion-4 a:active {
  color: #f4f5f6;
}

.emotion-4 a:hover {
  color: #fff;
}

.emotion-4 a:visited {
  color: #d8dcdf;
}

.emotion-4 .e16lo5rx0 {
  color: #000;
  background-color: #f4f5f6;
}

.emotion-4 .e16lo5rx0 * {
  color: inherit !important;
}

.emotion-4 .e16lo5rx0 svg {
  fill: #000;
}

.emotion-4 .e16lo5rx0:hover {
  background-color: #fff;
}

.emotion-4 .e16lo5rx0:hover * {
  color: #000 !important;
}

.emotion-4 .e16lo5rx0:hover svg {
  fill: #000;
}

.emotion-4 .etxwxxr0:link,
.emotion-4 .etxwxxr0:visited {
  color: #fff;
}

@media (min-width:768px) {
  .emotion-4 {
    width: 50%;
    box-shadow: 0 0.3rem 1rem 0 rgba(0,0,0,0.1);
    padding: calc(2.2222222222222223vw + 5.333333333333333px);
  }

@media (max-width:480px) {
    .emotion-4 {
      padding: 16px;
    }
}

@media (min-width:1200px) {
    .emotion-4 {
      padding: 32px;
    }
}
}

@media (min-width:768px) {
  section:nth-of-type(even) .emotion-4 {
    left: 50%;
  }
}

<div
  class="emotion-4 emotion-5"
  data-testid="Billboard-Content"
>
  <h2
    class="emotion-0 emotion-1"
  >
    Heading
  </h2>
  <h3
    class="emotion-2 emotion-3"
  >
    Subheading
  </h3>
  <p>
    Main Content
  </p>
</div>
`;
